src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -148,6 +149,46 @@ async function buildTxProposal(req, res) {
   }
 }
 
+/**
+ * Fetches all proposal data from the Atomic Swap Service for a specific identifier
+ */
+async function fetchFromService(req, res) {
+  if (!constants.SWAP_SERVICE_FEATURE_TOGGLE) {
+    res.status(405).json({ success: false, error: 'Method not allowed' });
+    return;
+  }
+
+  const { walletId } = req;
+  const { proposalId } = req.params;
+  const listenedProposals = await atomicSwapService.getListenedProposals(walletId);
+  if (!listenedProposals.has(proposalId)) {
+    res.status(404);
+    res.send({
+      success: false,
+      error: 'Proposal is not registered. Register it first through [POST] /register/:proposalId',
+    });
+    return;
+  }
+
+  const requestedProposal = listenedProposals.get(proposalId);
+  const { password } = requestedProposal;
+
+  // Fetching proposal from the service
+  try {
+    const serviceProposal = await atomicSwapService.serviceGet(proposalId, password);
+    res.json({ success: true, proposal: serviceProposal });
+  } catch (err) {
+    // TODO: Verify the status code instead of the message
```

Comment:
TODO is already done, right?

Reply:
You're right! Removed the obsolete comment..